pydicom/util/codify.py

```diff
@@ -185,6 +197,12 @@ def code_sequence(
     # Code line to create a new Sequence object
     if name_filter:
         seq_var = name_filter(seq_keyword)
+
+    name_count = var_names.count(seq_var)
+    var_names.append(seq_var)
+    orig_seq_var = seq_var
+    if name_count > 0:
+        seq_var += f"_{name_count}"
```

Comment:
I would move this into a local function, e.g.:
```py
    def sequence_name(name):
        # Append "_#" if name already in use
        name_count = var_names.count(name)
        var_names.append(name)
        if name_count > 0:
            name += f"_{name_count}"
        return name
```
and use it here and below (with `ds_name`)

Reply:
Done, but I kept the append to `var_names` in-line, I think it is easier to see the 'push' and the 'pop' easily in one place.

Reply:
That's ok, that was only a suggestion. As for the warning: I usually add an assert for mypy's sake in these cases, e.g.
```py
    def unique_name(name: str) -> str:
        assert var_names is not None
        name_count = var_names.count(name) - 1
        return name if name_count == 0 else name + f"_{name_count}"
```
This is not really nice, but the alternative I know (using a new variable) is error-prone, as you may use the old variable inadvertedly.